### PR TITLE
Roundlock, Lobbylock, FriendlyFire variables & fix AddHealth

### PR DESCRIPTION
**Pull request by SebasCapo**
Easier and more intuitive access to Round-lock, Lobby-lock & Friendly-fire booleans.

### CODE REVIEW

**Comment by js6pak** [comment]
Can't you just use properties instead of Get-/Set- methods? Is it even needed? Also incorrect formatting

**Comment by galaxy119** [comment]
For extensions that change something that aren't instanced (ie, static variables/etc), it's prefered for you to use a get/set.
Something like (pseudo code below)
```
public static bool FriendlyFire
{
    get
    {
        return ServerConsole.FriendlyFire;
    };

    set
    {
        ServerConsole.FriendlyFire = value;
    }
}```

**Comment by SebasCapo** [comment]
Ah, I see, makes sense! I'mma make another commit later with those changes!

**Comment by js6pak** [comment]
Please fix formatting :< (what are you using for editing code? notepad?)

**Comment by SebasCapo** [comment]
I have to edit the github file :( Can't access my pc rn

File: EXILED_Main/Extensions/Player.cs [hunk 1]
```diff
@@ -603,8 +603,12 @@ public static Room GetCurrentRoom(this ReferenceHub player)
 		/// </summary>
 		/// <param name="player"></param>
 		/// <param name="amount"></param>
-		public static void AddHealth( this ReferenceHub player, float amount ) =>
-			player.playerStats.health = Mathf.Clamp(amount, 1, player.playerStats.maxHP);
+		public static void AddHealth(this ReferenceHub player, float amount, bool capAtMax = false) {
```
Comment (js6pak) [inline EXILED_Main/Extensions/Player.cs:606]: Won't changing signature break old plugins?

Comment (SebasCapo) [reply EXILED_Main/Extensions/Player.cs:606]: AFAIK it shouldn't 🤔, it'll just set capAtMax to false by default, but could be wrong

File: EXILED_Main/Extensions/Player.cs [hunk 2]
```diff
@@ -603,8 +603,12 @@ public static Room GetCurrentRoom(this ReferenceHub player)
...
+			if(capAtMax) {
+				if(player.playerStats.health + amount > player.playerStats.maxHP) amount = player.playerStats.maxHP - player.playerStats.health;
```
Comment (js6pak) [inline EXILED_Main/Extensions/Player.cs:608]: Whats the point of doing that?

Comment (SebasCapo) [reply EXILED_Main/Extensions/Player.cs:608]: MaxHp cap :P

File: EXILED_Main/Extensions/Player.cs (hunk 1 above)
Comment (js6pak) [reply EXILED_Main/Extensions/Player.cs:606]: Still signature is changed, so plugins compiled with old version of EXILED will break

**Comment by js6pak** [comment]
And AddHealth fix couldn't be just simply `player.playerStats.health = Mathf.Clamp(player.playerStats.health + amount, 1, player.playerStats.maxHP);`?

**Comment by SebasCapo** [comment]
Uh, sure, I didn't really know how to use Mathf.Clamp, if that works I'd rather use it

**Comment by SebasCapo** [comment]
Alright, should be better now!

**Comment by js6pak** [comment]
`amount += player.playerStats.health; player.playerStats.health = capAtMax ? Mathf.Min(amount, player.playerStats.maxHP) : amount` would be shortest

**Comment by galaxy119** [comment]
For setting FriendlyFire - 
Firstly, you need to set NetworkfriendlyFire, not friendlyFire

Secondly instead of doing a GetComponentsOfType<> why not 
```
foreach (ReferenceHub hub in Player.GetHubs())
    hub.SetFriendlyFire(value);
``` 
?

**Comment by SebasCapo** [comment]
Damn, you're right, that's better! Updated with that method! ^^

**Comment by FruitBoi** [comment]
Looks good.

**Pull request merged**